examples/wms-custom-tilegrid-512x256.js

```diff
@@ -25,7 +25,6 @@ var layers = [
     source: new ol.source.MapQuest({layer: 'sat'})
   }),
   new ol.layer.Tile({
-    extent: [-13884991, 2870341, -7455066, 6338219],
```

Comment:
I understand. We used to have an extent option on ol.source.Tile to just prevent requesting tiles outside that extent. So your patch adds this feature back, except that the extent option is on the tile grid rather the source.